docs/guide/start-identity-permissions-and-scopes.md

```diff
@@ -0,0 +1,201 @@
+   authorization code for verification.
+
+5. The Authorization Server (Yii2-Oauth2-Server) authenticates the Client, validates the authorization code, and ensures
+   that the redirection URI received matches the URI used to redirect the Client in step (3).  
+   If valid, the Authorization Server (Yii2-Oauth2-Server) responds back with an access token and, optionally, a 
+   refresh token.
+
+### 2-legged OAuth 2.0 flow
+
+The Client credentials Grant is used typically when the client is acting on its own behalf (server-to-server).
+
+Client credentials flow:
+```
+     +---------+                                  +---------------+
+     |         |                                  |               |
+     |         |>--(1)- Client Authentication --->| Authorization |
+     | Client  |                                  |     Server    |
+     |         |<--(2)---- Access Token ---------<| (Yii2-Oauth2- |
+     |         |                                  |    Server)    |
+     +---------+                                  +---------------+
+```
+
+1. The Client authenticates (via its 'identifier' and 'secret') with the Authorization Server (Yii2-Oauth2-Server) and
+   requests an access token from the token endpoint.
+
+2. The Authorization Server (Yii2-Oauth2-Server) authenticates the Client, and if valid, issues an access token.
+
+Since the client authentication is used as the authorization grant, no additional authorization request is needed.
+
+
+Yii2-Oauth2-Server Integration
+------------------------------
+
+The Yii2-Oauth2-Server integrates with the Yii2 application by taking care of the entire Oauth 2.0 flow.
+After successful authorization, when the Client makes server requests with a valid authorization header
+the Yii2-Oauth2-Server will set the Yii2 "user identity class" (`Yii::$app->user->identity`) for the "user component"
+(`Yii::$app->user`). 
+
+### Identity and RBAC usage
+
+Let's say that in our example there is a backend application (from now on abbreviated as BEA) has 2 roles:
+'user' and 'admin'.
+When a User authorizes a Client, let's say a frontend application(from now on abbreviated as BEA), to use the BEA,
+all calls that are made by that Client will be automatically (under the condition that the Client correctly sends the 
+"authentication" header) done with that user's identity.
+
+Let's say that whe have two users: Alice is "admin" and user Bob is a regular "user".
+
+When Alice authorizes the FEA via the Authorization Code Grand the FEA can use the "bearer token" authorization header
+to make API calls against the BEA. For all these calls the `Yii::$app->user->identity` will be set to the user identity
+of Alice. When Bob would authorize the FEA all calls with his bearer token by the FEA will set the
+`Yii::$app->user->identity` to Bob's identity.
+
+Since the identity is set, you can simply use `Yii::$app->user->can('role_you_want_to_check')` anywhere you want.
```

Comment:
Passing correct bearer token ensures that correct user identity information is set in the user identity component as explained above. Since the identity is set, you can check if user is allowed  access certain resources like you do in any Yii application, e.g. `Yii::$app->user->can('role_you_want_to_check')` anywhere you want. See Yii2 Guide on Authorization for more details on this.

Reply:
Improved based on your feedback.